ngraph/core/reference/include/ngraph/runtime/reference/max_pool.hpp

```diff
@@ -110,6 +110,340 @@ void max_pool(const T* arg,
     }
     NGRAPH_SUPPRESS_DEPRECATED_END
 }
+
+namespace {
+void validate_max_pool_kernel_params(const size_t dims,
+                                     const Shape& kernel,
+                                     const Strides& kernel_strides,
+                                     const Strides& kernel_dilations,
+                                     const Shape& pads_begin,
+                                     const Shape& pads_end) {
+    NGRAPH_CHECK(kernel.size() == dims && kernel_strides.size() == dims && kernel_dilations.size() == dims &&
+                     pads_begin.size() == dims && pads_end.size() == dims,
+                 "One of the MaxPool params does not match the ",
+                 dims,
+                 "D implementation.\nkernel=",
+                 kernel,
+                 "\nkernel_strides=",
+                 kernel_strides,
+                 "\nkernel_dilations=",
+                 kernel_dilations,
+                 "\npads_begin=",
+                 pads_begin,
+                 "\npads_end=",
+                 pads_end);
+}
+
+template <typename T>
+struct Coord : public std::vector<T> {
```

Comment:
@tomdol I think we need to provide more documentation for this code, can you do it in the separate PR?